fakesnow/transforms.py

```diff
@@ -169,6 +169,29 @@ def drop_schema_cascade(expression: exp.Expression) -> exp.Expression:
     return new
 
 
+def dateadd_string_literal_timestamp_cast(expression: exp.Expression) -> exp.Expression:
+    """Snowflake's DATEADD function implicitly casts string literals to
+    timestamps regardless of unit.
+    """
+    if not isinstance(expression, exp.DateAdd):
+        return expression
+
+    if not isinstance(expression.this, exp.Literal) or not expression.this.is_string:
```

Comment:
NB: It's a big future feature, but at some point if we had all the column lineage exposed, we could support column expressions that were VARCHAR too.

Reply:
that might be helpful on timestamp casting as well. snowflake's `cast( ... timestamp)` and `to_timestamp` both support conversion from numeric and varchar values, but on duckdb `cast` is solely for varchar and `to_timestamp` for numeric/bigint.